research/src/pages/component-spec-template.mdx

```diff
@@ -0,0 +1,213 @@
+> This document represents a component specification template. Its purpose
+> is to provide a point-in-time guide as to how a component should be
+> authored, in light of common anatomies, behaviors, and accessibility.
+> The primary audience is a developer authoring a given component.
+
+# {ComponentName} Component Specification
+
+## Overview
+
+*The name of the component, along with a high-level description.*
+
+### Background
+
+*Relevant historical or background information, related existing issues, etc.*
+
+### Use Cases
+
+*Primary use cases for this component.*
+
+### Non-goals
+
+*A list of use cases, features, or functionality which are **not** goals for the component.*
+  
+### Features
+
+*A list of the key features unique to this component.*
+
+### Risks and Challenges
+
+*Notable risks or challenges associated with implementing or shipping the component.*
+
+### Prior Art/Examples
+
+*Screenshots and/or links to existing, canonical, or exemplary implementations of the component.*
+
+---
+
+## Design
+
+*Describe the design of the component, thinking through several perspectives:*
+
+- *A customer using the component on a web page.*
+- *A developer building an app with the component and interacting through HTML/CSS/JavaScript.*
+- *A designer customizing the component.*
+
+### API
+
+*Outline the key elements of the component's public API surface, taking into consideration conformance guidelines. Make sure to discuss differences and rationalizations. Consider high and low-level APIs. Attempt to design a powerful and extensible low-level API with a high-level API for developer/designer ergonomics and simplicity.*
+
+#### Properties and Attributes
+
+*Example Table*
+
+| Property Name | Attribute Name | Type | Default Value | Description |
+| ------------- | -------------- | ---- | ------------- | ----------- |
+| `value` | `value` | `number` | Value of `min` | The current value of the slider. |
+| `min` | `min` | `number` | `0` | The minimum value of the slider. |
+
+#### Methods
+
+*Example Table*
+
+| Signature | Description |
+| ----------- | ---------- |
+| `increment(value: number = this.step): void` | Increments the `value` of the component by the amount specified by `step`, clamped within `min`/`max` values. |
+| `decrement(value: number = this.step): void` | Decrements the `value` of the component by the amount specified by `step`, clamped within `min`/`max` values. |
+
+#### Events
+
+*Example Table*
+
+| Event Name | Detail Type | Bubbles | Composed | Cancellable | Dispatch Behavior |
+| ---------- | ----------- | ------- | -------- | ----------- | ----------------- |
+| `change` | none | `true` | `true` | `false` | Fired when the slider's `value` changes. |
+
+### Appearance
+
+*Screenshots and/or description of the basic appearance of the component.*
+
+### Anatomy
+
+*Outline the component's structure with a diagram of its visual tree (shadow dom). Enumerate key areas of visual customization.*
```

Comment:
```suggestion
*Outline the component's structure with a diagram of its visual tree. Enumerate key areas of visual customization.*
```

I think referencing "shadow dom" gets further into the implementation specifics than we want to get.

Reply:
I've reworded this sentence to remove that reference and be a bit more general.